scripts/parse_performance_metrics.py

```
@@ -0,0 +1,200 @@
+
+def _extract_metrics_from_message(message: str) -> Dict[str, Any] | None:
+    sentinel_index = message.find(_METRICS_SENTINEL)
+    if sentinel_index == -1:
+        return None
+    payload = message[sentinel_index + len(_METRICS_SENTINEL) :].strip()
+    if not payload:
+        return None
+    metrics = _try_decode_json(payload)
+    if metrics is not None:
+        return metrics
+    # Some CI logs escape the JSON payload – attempt a secondary decode.
```

Ruff (0.14.0)

Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)